src/crh/mod.rs

```diff
@@ -14,12 +14,29 @@ pub mod constraints;
 #[cfg(feature = "r1cs")]
 pub use constraints::*;
 
-pub trait FixedLengthCRH {
+pub trait CRH {
     const INPUT_SIZE_BITS: usize;
 
     type Output: ToBytes + Clone + Eq + core::fmt::Debug + Hash + Default;
     type Parameters: Clone + Default;
 
-    fn setup<R: Rng>(r: &mut R) -> Result<Self::Parameters, Error>;
+    fn setup_crh<R: Rng>(r: &mut R) -> Result<Self::Parameters, Error>;
```

Comment:
Why rename this? Usually it'll be invoked as `CRH::setup`, so it's not ambiguous, no?

Reply:
I changed it back to `setup`